;; An example failing computation, to exercise debugging introspection.
;; This turned into essentially a repeat of test-continuations.scm.

(import ((use 'traceback) (export out))
  on-error-traceback)

(to (factorial n)
  (if (= 0 n)
      (error "I don't know 0!")
      (* n (factorial (- n 1)))))

(to (complain-and-resume sequel evil)
  (on-error-traceback sequel evil)
  (out .say "About to resume, pretending 0! = 10\n")
  (sequel .answer 10))

(with-signal-handler complain-and-resume
                     (: (out .print (factorial 5))))
